main.js

```diff
@@ -0,0 +1,430 @@
+        return putPieceInRow;
+    };
+
+
+// 2x2 SQUARE BONUS
+    function squareBonusCheck(boardArrayClick) {
+        if (boardArray[row - 1][column + 1] === 'X') {
+            if (boardArray[row - 0][column + 1] === 'X' && boardArray[row - 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS top right');
+                return;
+            }
+            ;
+        }
+        ;
+        if (boardArray[row - 1][column - 1] === 'X') {
+            if (boardArray[row - 0][column - 1] === 'X' && boardArray[row - 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS top left');
+                return;
+            }
+            ;
+        }
+        ;
+        if (boardArray[row + 1][column - 1] === 'X') {
+            if (boardArray[row - 0][column - 1] === 'X' && boardArray[row + 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS bottom left');
+                return;
+            }
+            ;
+        }
+        ;
+        if (boardArray[row + 1][column + 1] === 'X') {
+            if (boardArray[row - 0][column + 1] === 'X' && boardArray[row + 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS bottom right');
+                return;
+            }
+            ;
+        }
+        ;
+    };
+
+    function winCondition(row, column, array, player) {
+        // console.log('chickens');
+        // console.log('row', row);
+        // console.log('column', column)
+        // console.log('array', array)
+        // console.log('player:', player);
+        row = Number(row);
+        column = Number(column);
+        if (array[row][column].token !== player) {
+            return false
+        }
+        var row_direction_array = [0, -1, -1, -1];
+        var column_direction_array = [-1, 1, 0, -1];
+
+        for (var x = 0; x < row_direction_array.length; x++) {
+            var row_direction = row_direction_array[x];
+            var column_direction = column_direction_array[x];
+
+            var check_row = row + row_direction;
+            var check_column = column + column_direction;
+            var inline_counter = 1;
+
+            for (var i = 0; i < 2; i++) {
+                if (array[check_row] === undefined) {
+                }
+                else if (array[check_row][check_column] === undefined) {
+                }
+                else {
+                    while (array[check_row][check_column].token === player) {
+                        inline_counter += 1;
+                        check_row += row_direction;
+                        check_column += column_direction;
+                        if (array[check_row] === undefined) {
+                            break
+                        }
+                        else if (array[check_row][check_column] === undefined) {
+                            break
+                        }
+                    }
+                }
+                row_direction *= -1;
```

Comment:
You notice here how you'll wind up doing this same basic thing multiple times, but you give it different row direction and column_directions?  You could have made this a function and pass in the row and column direction into it, and simplified your code.  Like this : 

```
function getValueAtSpot(col, row){
   if(array[row] === undefined || array[row][col] === undefined){
      return false;
   } 
   return array[row][col];
}
function checkInDirection( startCol, startRow, colDirection, RowDirection, player){
    var inline_counter = 0;
    var token = getValueAtSpot(startCol, startRow);
    if( !token ){
      while(  token = getValueAtSpot(startCol, startRow) ){
          inline_counter+=1;
          startCol+=colDirection;
          startRow+=rowDirection;
      }
      if(inline_counter>=4){
           return true;
      }
      return false;
   }
```